### PR TITLE
Fixed healthchecks for CNI-enabled agents.

### PR DESCRIPTION
**Pull request by zed3250**
JIRA issues: MESOS-4614

### CODE REVIEW

**Review by kamaradclimber — CHANGES_REQUESTED** [review]
It's not clear for a review (and for your future self) why this change fixes the issue (nor what what the issue was).

Could you detail them in the commit message?

**Comment by Lqp1** [comment]
Can you run marathon-test-suite locally with that version of marathon to make sure there is no regression when not using CNI on mesos?

File: src/test/scala/mesosphere/marathon/core/task/TaskTest.scala [hunk 1]
```diff
@@ -97,14 +97,14 @@ class TaskTest extends UnitTest with Inside {
       f.taskWithMultipleNetworksAndOneIp.status.networkInfo.effectiveIpAddress(f.appWithIpAddress).value should equal(f.ipString1)
     }
 
-    "effectiveIpAddress returns None if there is no ip" in {
+    "effectiveIpAddress returns agent hostname if there is no ip" in {
       val f = new Fixture
-      f.taskWithMultipleNetworkAndNoIp.status.networkInfo.effectiveIpAddress(f.appWithIpAddress) should be(None)
+      f.taskWithMultipleNetworkAndNoIp.status.networkInfo.effectiveIpAddress(f.appWithIpAddress) should be(Some("agent1.mesos"))
```
Comment (Lqp1) [inline src/test/scala/mesosphere/marathon/core/task/TaskTest.scala:102]: Looks an important change; I think marathon was waiting for None in many parts of the code (and that's how it behave without CNI). Have you checked where effectiveIpAddress is used to make sure there is no other side effect than in the tests?

Comment (zed3250) [reply src/test/scala/mesosphere/marathon/core/task/TaskTest.scala:102]: There's always going to be an effective IP (if the CNI is off then it'll be the host's IP) so it's just to fix the tests

**Review by Lqp1 — APPROVED** [review]
Let's squash and push on the branch now

**Pull request merged**